### ISSUE TITLE
Cluster serving: the heartbeat of TaskManager with id * timed out

### ISSUE THREAD

**Issue by cwpeng-cn** [intel-analytics/analytics-zoo#2721]
## Information
 **config.yaml:**

```
## Analytics-zoo Cluster Serving

model:
  # model path must be provided
  # path: /dev/workdir/cwpeng/models/IntelModel/person-detection-retail-0002
  path: /dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003
  # mode, default is recommended, if error is raised, set single
  mode:
  # the inputs of the tensorflow model, separated by ","
  inputs:
  # the outputs of the tensorflow model, separated by ","
  outputs:
data:
  # default, localhost:6379
  src:
  # shape must be provided, e.g, [[1,2],[3,224,224],[3,4]], if only one input brackets could be omitted
  shape: 3,224,224
  # type must be provided, e.g, [image, tensor, sparse_tensor, tensor], if only one input brackets could be omitted
  type: image
  # default, topN(1)
  filter:
params:
  # default, 4
  core_number: 1
  # default: OFF
  performance_mode:
redis:
  # default, 4g
  maxmem:
spark:
  # default, local[*]
  master:
  # default, 4g
  driver_memory:
  # default, 1g
  executor_memory:
  # default, 1
  num_executors:
  # default, 4
  executor_cores:
  # default, 4
  total_executor_cores:
secure:
  secure_enabled:
```
**The OpenVINO model info:**
https://docs.openvinotoolkit.org/latest/omz_models_intel_emotions_recognition_retail_0003_description_emotions_recognition_retail_0003.html

---

## Error

Then I start cluster-serving and the Flink UI shows that the task is running. But when I try to write data to redis, the service hangs and the error message is as follows.
`Caused by: java.util.concurrent.TimeoutException: The heartbeat of TaskManager with id b016fa716b23d781e8de46b9352bf232  timed out.`

---

## Debug
Then I did some attempts to figure it out.
**1.** 

First, I run cluster-serving.scala in IDEA and feed data. An error has occurred, the log is as follows：
```
(Loading config at ,/dev/workdir/cwpeng/demo/config.yaml)
{path=/dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003, mode=null, inputs=null, outputs=null}path getted: /dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003
{src=null, shape=3,64,64, type=image, filter=null}shape getted: 3,64,64
{src=null, shape=3,64,64, type=image, filter=null}type getted: image
{core_number=1, performance_mode=null}core_number getted: 1
loading params, time is 08/04/2020 09:19:30
OpenvinoNativeLoader load OpenVINO 2020_R2 succeed !
2020-08-13 16:24:41 INFO  InferenceSupportive$:45 - load native so for OpenVINO time elapsed [0 s, 203 ms].
2020-08-13 16:24:41 INFO  InferenceSupportive$:45 - load OpenVINO IR time elapsed [0 s, 43 ms].
Running Cluster Serving in test mode with parallelism 1
2020-08-13 16:24:44 INFO  Slf4jLogger:92 - Slf4jLogger started
2020-08-13 16:24:44 INFO  Slf4jLogger:92 - Slf4jLogger started
Cluster Serving Flink job id written to file.
2020-08-13 16:24:46 INFO  FlinkRedisSink:48 - FlinkRedisSink connect to plain Redis successfully.
2020-08-13 16:24:47 INFO  FlinkRedisSource:54 - FlinkRedisSource connect to plain Redis successfully.
redis.clients.jedis.exceptions.JedisDataException: BUSYGROUP Consumer Group name already exists exist group
Inference new batch ..................
This batch size is 1
Batch input (copy, resize) time 0.012633085 s
2020-08-13 16:24:48 INFO  OpenVINOModel$:42 - Prepare OpenVinoInferenceSupportive
2020-08-13 16:24:48 INFO  OpenVinoInferenceSupportive$:129 - Force native loader
2020-08-13 16:24:48 INFO  OpenVINOModel$:49 - Lazy loading OpenVINO model
2020-08-13 16:24:48 INFO  OpenVINOModel$:63 - Load fp32 model
[ INFO ] InferenceEngine: 0x7f8cdbffc030
Loading plugin
[ INFO ] Loading network files:
	/tmp/OpenVINO4035266677198283818.xml
	/tmp/OpenVINO6566739287632769430.bin
[ INFO ] Network batch size is 1
[ INFO ] Input Layout NCHW
[ INFO ] Input Precision FP32
[ INFO ] Loading model to the device
double free or corruption (!prev)
```
The above error message is too vague, it is difficult for us to locate the fault.

**2.** 

In order to verify whether the problem is caused by flink, I call the zoo interface. The code is as follows：
```
val model = new InferenceModel()
model.doLoadOpenVINO("/dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003/emotions-recognition-retail-0003.xml", "/dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003/emotions-recognition-retail-0003.bin")
val predictOutput = model.doPredict(Tensor[Float](1,1,3,224, 224))
print(predictOutput)
```

An error also occurred, the information is as follows:
```
2020-08-13 16:33:27 WARN  InferenceSupportive$:269 - concurrentNum is 20 > 1, openvino model does not support shared weights model copies
OpenvinoNativeLoader load OpenVINO 2020_R2 succeed !
2020-08-13 16:33:28 INFO  InferenceSupportive$:45 - load native so for OpenVINO time elapsed [0 s, 189 ms].
2020-08-13 16:33:28 INFO  InferenceSupportive$:45 - load OpenVINO IR time elapsed [0 s, 60 ms].
2020-08-13 16:33:29 INFO  OpenVINOModel$:42 - Prepare OpenVinoInferenceSupportive
2020-08-13 16:33:29 INFO  OpenVinoInferenceSupportive$:129 - Force native loader
2020-08-13 16:33:29 INFO  OpenVINOModel$:49 - Lazy loading OpenVINO model
2020-08-13 16:33:29 INFO  OpenVINOModel$:63 - Load fp32 model
[ INFO ] InferenceEngine: 0x7f38ebcc4030
Loading plugin
[ INFO ] Loading network files:
	/tmp/OpenVINO1195750505126884846.xml
	/tmp/OpenVINO5420391118679213992.bin
[ INFO ] Network batch size is 1
[ INFO ] Input Layout NCHW
[ INFO ] Input Precision FP32
[ INFO ] Loading model to the device
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f38eba20926, pid=17346, tid=0x00007f38e3204700
#
# JRE version: OpenJDK Runtime Environment (8.0_252-b09) (build 1.8.0_252-8u252-b09-1~18.04-b09)
# Java VM: OpenJDK 64-Bit Server VM (25.252-b09 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [libinference_engine.so+0x1c1926]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /dev/workdir/official/analytics-zoo/hs_err_pid17346.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
```

I consulted @qiyuan, and qiyuan told me that the error was caused by the mismatch in size.
Then I modified the input and got the correct result.
```val model = new InferenceModel()
model.doLoadOpenVINO("/dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003/emotions-recognition-retail-0003.xml", "/dev/workdir/cwpeng/models/IntelModel/emotions-recognition-retail-0003/emotions-recognition-retail-0003.bin")
# val predictOutput = model.doPredict(Tensor[Float](1,1,3,224, 224))
val predictOutput = model.doPredict(Tensor[Float](1,1,3,64, 64))
print(predictOutput)
```

**So the size mismatch will cause an error(`the heartbeat of TaskManager with id * timed out`).  It is difficult for users to find the reason for this error. Can we add some hints to make this easier?**.

**Comment by qiyuangong** [comment]
Hi @cwpeng-cn 

I think you can split this issue into 2 issue. This issue for heartbeat, and another issue for input shape check and warning. So, we can track and fix separately. :)

**Comment by cwpeng-cn** [comment]
> Hi @cwpeng-cn
> 
> I think you can split this issue into 2 issue. This issue for heartbeat, and another issue for input shape check and warning. So, we can track and fix separately. :)

I think the size mismatch caused JNI to report an error. This is very low-level. What the user can see is that flink reports a heartbeat timeout error without other information.

**Comment by glorysdj** [comment]
I think the OpenVINOModel should catch the error/exception and wrap it as a Java RuntimeException with clearer message so the fink TimeoutException can get clearer root cause.

**Comment by qiyuangong** [comment]
> I think the OpenVINOModel should catch the error/exception and wrap it as a Java RuntimeException with clearer message so the fink TimeoutException can get clearer root cause.

Yes. I have opened a zoo-core issue (https://github.com/intel-analytics/analytics-zoo-core/issues/99) for this problem. Will submit a PR for that.
